SDL/Module.h

```diff
@@ -441,83 +441,32 @@ namespace SDL
-            std::vector<unsigned int> connectedModuleDetIds_pLStoLayer1Subdet4_pos = SDL::moduleConnectionMap_pLStoLayer1Subdet4_pos.getConnectedModuleDetIds(isuperbin);
-            std::vector<unsigned int> connectedModuleDetIds_pLStoLayer2Subdet4_pos = SDL::moduleConnectionMap_pLStoLayer2Subdet4_pos.getConnectedModuleDetIds(isuperbin);
-            std::vector<unsigned int> connectedModuleDetIds_pLStoLayer3Subdet4_pos = SDL::moduleConnectionMap_pLStoLayer3Subdet4_pos.getConnectedModuleDetIds(isuperbin);
-            std::vector<unsigned int> connectedModuleDetIds_pLStoLayer4Subdet4_pos = SDL::moduleConnectionMap_pLStoLayer4Subdet4_pos.getConnectedModuleDetIds(isuperbin);
-            connectedModuleDetIds_pos.insert(connectedModuleDetIds_pos.end(),connectedModuleDetIds_pLStoLayer1Subdet5_pos.begin(),connectedModuleDetIds_pLStoLayer1Subdet5_pos.end());
-            connectedModuleDetIds_pos.insert(connectedModuleDetIds_pos.end(),connectedModuleDetIds_pLStoLayer2Subdet5_pos.begin(),connectedModuleDetIds_pLStoLayer2Subdet5_pos.end());
-            connectedModuleDetIds_pos.insert(connectedModuleDetIds_pos.end(),connectedModuleDetIds_pLStoLayer3Subdet5_pos.begin(),connectedModuleDetIds_pLStoLayer3Subdet5_pos.end());
-            connectedModuleDetIds_pos.insert(connectedModuleDetIds_pos.end(),connectedModuleDetIds_pLStoLayer1Subdet4_pos.begin(),connectedModuleDetIds_pLStoLayer1Subdet4_pos.end());
-            connectedModuleDetIds_pos.insert(connectedModuleDetIds_pos.end(),connectedModuleDetIds_pLStoLayer2Subdet4_pos.begin(),connectedModuleDetIds_pLStoLayer2Subdet4_pos.end());
-            connectedModuleDetIds_pos.insert(connectedModuleDetIds_pos.end(),connectedModuleDetIds_pLStoLayer3Subdet4_pos.begin(),connectedModuleDetIds_pLStoLayer3Subdet4_pos.end());
-            connectedModuleDetIds_pos.insert(connectedModuleDetIds_pos.end(),connectedModuleDetIds_pLStoLayer4Subdet4_pos.begin(),connectedModuleDetIds_pLStoLayer4Subdet4_pos.end());
-
             int sizes_pos = 0;
-            sizes_pos += connectedModuleDetIds_pLStoLayer1Subdet5_pos.size();
-            sizes_pos += connectedModuleDetIds_pLStoLayer2Subdet5_pos.size();
-            sizes_pos += connectedModuleDetIds_pLStoLayer3Subdet5_pos.size();
-            sizes_pos += connectedModuleDetIds_pLStoLayer1Subdet4_pos.size();
-            sizes_pos += connectedModuleDetIds_pLStoLayer2Subdet4_pos.size();
-            sizes_pos += connectedModuleDetIds_pLStoLayer3Subdet4_pos.size();
-            sizes_pos += connectedModuleDetIds_pLStoLayer4Subdet4_pos.size();
+            for ( auto const& mCM_pLS : moduleConnectionMap_pLStoLayer_pos ) {
+                std::vector<unsigned int> connectedModuleDetIds_pLS = mCM_pLS.getConnectedModuleDetIds(isuperbin);
+                connectedModuleDetIds_pos.insert(connectedModuleDetIds_pos.end(),connectedModuleDetIds_pLS.begin(),connectedModuleDetIds_pLS.end());
+                sizes_pos += connectedModuleDetIds_pLS.size();
```

Comment:
Is this a typo? You use connectedModuleDetIds_pLS for all three sizes (pos, neg, standard).

Reply:
No, because the variable used for the sizes is a local variable to each loop, initialized two lines above in this case, so it doesn't carry over from loop to loop. However, I see where the confusion is coming from, so I updated the name in each loop with a descriptive suffix ([71ba876](https://github.com/SegmentLinking/TrackLooper/pull/343/commits/71ba8769eaa0ec8b6ca06cbc0e4ec312751d8271)).